### PR TITLE
fix(cloudfront): fromOriginAccessIdentityName is a misnomer

### LINKED ISSUE #20141: @aws-cdk/cloudfront: OriginAccessIdentity mixes up id and name
### Describe the bug

`OriginAccessIdentity.fromOriginAccessIdentityName` asks for the OAI __name__, but actually needs the OAI __id__.

Set up the following bucket with a Cloudfront OAI __name__ from a different account:

```
const bucket = new Bucket(this, "Bucket");
const originAccessIdentity =
  OriginAccessIdentity.fromOriginAccessIdentityName(
    this,
    "OriginAccessIdentity",
    "<add-OAI-name-here>"
  );

bucket.grantRead(originAccessIdentity);
```

Then deploy.

### Expected Behavior

The deployment should work.

### Current Behavior

The deployment fails because what is needed is actually the OAI id, not the OAI name. This makes `OriginAccessIdentity.fromOriginAccessIdentityName` quite a misnomer.

### Reproduction Steps

See above.

### Possible Solution

Rename the method and internal properties to `OriginAccessIdentity.fromOriginAccessIdentityId`.

### Additional Information/Context

Not sure if this bug is CDK-specific or a general CloudFormation problem.

### CDK CLI Version

2.20.0

### Framework Version

_No response_

### Node.js Version

16

### OS

Windows 11

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by dbartholomae**
fixes #20141

----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by dbartholomae** [comment]
Fixes #20141

**Comment by dbartholomae** [comment]
I had to take two shortcuts:
1. I didn't add a test for the `originAccessIdentityId` property as I also didn't find a test for `originAccessIdentityName` and couldn't find a good way to test it that goes over what TypeScript already guarantees (that it is a non-empty string).
2. I had to remove the test for `fromOriginAccessIdentityName` as calling deprecated functions from a test currently is an error. It could make sense to deactivate this check or at least add some way to deactivate it inside a test.

For both, I'm happy about suggestions, but if CI is green, then from my point of view, the PR can already be merged as-is.

**Review by peterwoodworth — DISMISSED** [review]
Like you said, It would be nice to verify that the deprecated prop works. We have a `testDeprecated` which behaves similarly to `test`. Check out an example of this here https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-elasticsearch/test/domain.test.ts

**Comment by dbartholomae** [comment]
I've added a test for the now deprecated `fromOriginAccessIdentityName` (for now as a fixup for easier review, will rebase once the PR is approved from your side), but I'm not sure how to best write tests for the properties. I could test that they exist and are strings - but this is actually already assured by TypeScript. And for the exact content I'm not sure how to predict this, as it contains a random letter-number-combination if I'm not mistaken.

**Comment by peterwoodworth** [comment]
I think the tests you have here are plenty sufficient @dbartholomae,

I'll give this an approval once you make your desired rebase

**Comment by dbartholomae** [comment]
@peterwoodworth here you are :)

**Review by peterwoodworth — APPROVED** [review]
LGTM! going to let another team member take a quick peek to double check that deprecation was handled properly here, then it should be merged

**Review by kaizencc — COMMENTED** [review]
The deprecation here looks fine @peterwoodworth, @dbartholomae. One call out for something weird in the Cloudformation docs -- not sure why a return value exists in the docs and is also "not supported".

File: packages/@aws-cdk/aws-cloudfront/lib/origin-access-identity.ts [hunk 1]
```diff
@@ -93,11 +126,23 @@ export class OriginAccessIdentity extends OriginAccessIdentityBase implements IO
...
 
   /**
-   * The Origin Access Identity Name (physical id)
+   * The Origin Access Identity Id (physical id)
+   * It is misnamed and superseded by the correctly named originAccessIdentityId
+   *
+   * @attribute
+   * @deprecated use originAccessIdentityId instead
+   */
+  public get originAccessIdentityName() {
```
[the remark this answers is not in the document]
Comment (corymhall) [reply packages/@aws-cdk/aws-cloudfront/lib/origin-access-identity.ts:135]: Or if they would support it in the future?

[the remark this answers is not in the document]
Comment (peterwoodworth) [reply packages/@aws-cdk/aws-cloudfront/lib/origin-access-identity.ts:135]: I tested it and it's currently supported if I create an OAI ID and reference the attribute in a CfnOutput. Seems to just work

[the remark this answers is not in the document]
Comment (peterwoodworth) [reply packages/@aws-cdk/aws-cloudfront/lib/origin-access-identity.ts:135]: going to remove the label and let this merge

[the remark this answers is not in the document]
Comment (peterwoodworth) [reply packages/@aws-cdk/aws-cloudfront/lib/origin-access-identity.ts:135]: correction - we just use ref here which *does* return the OAI Id. But, using the ID attribute still works will return the exact same value as ref so...

**Pull request merged**